arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -663,7 +663,10 @@ std::unique_ptr<ReplicationIterator> RocksDBCollection::getReplicationIterator(
 ///////////////////////////////////
 
 Result RocksDBCollection::truncate(transaction::Methods& trx, OperationOptions& options) {
-  ::TruncateTimeTracker timeTracker(_statistics._rocksdb_truncate_usec, _statistics, options);
+
+  if (_statistics._exportReadWriteMetrics) {
+    ::TruncateTimeTracker timeTracker(_statistics._rocksdb_truncate_sec->get(), _statistics, options);
```

Comment:
This will not work. This creates a new block scope with the TimeTracker, which will directly be destroyed again as the block is left instantly. So this will **not** report the time the operation has taken, but only the time spent in the `if` block.